modules/aws/vpc_test.go

```diff
@@ -61,6 +61,71 @@ func TestGetFirstTwoOctets(t *testing.T) {
 	}
 }
 
+func TestIsPublicSubnet(t *testing.T) {
```

Comment:
I'd also like to see:

- Associating a route for NAT gateway and verifying it is not labeled as public
- Associating a IGW route that is limited in scope (NOT `0.0.0.0/0`) for the destination

Since creating these routes gets fairly complex, it would be nice to have a top level terraform module to create these test scenarios. You can create a single VPC with multiple subnets and multiple routes, and then test each subnet with the function. That way, we can rely on Terraform to handle the create and destroy logic which has complex interleaving dependencies.

Reply:
Not sure where I could implement that... Should I add the terraform scripts to `/examples` and the test like a regular terratest to `/test`?

Reply:
Sorry I should have been clearer. That is exactly what I am suggesting.

Reply:
done